src/worldsync.js

```diff
@@ -0,0 +1,149 @@
+const Vec3 = require('vec3').Vec3
+const EventEmitter = require('events').EventEmitter
+
+function posInChunk (pos) {
+  return new Vec3(Math.floor(pos.x) & 15, Math.floor(pos.y), Math.floor(pos.z) & 15)
+}
+
+class WorldSync extends EventEmitter {
+  constructor (world) {
+    super()
+    this.async = world
+  }
+
+  initialize (iniFunc, length, width, height = 256, iniPos = new Vec3(0, 0, 0)) {
```

Comment:
I kept a duplicated of this because actually this is quite different : this is using the sync primitive and might fail (do nothing) if columns are not loaded.
Also it might be much faster than the async version

Reply:
ok